### DISCUSSION TITLE
It seems that preserve_notate_commends only retains some comments instead of all comments

### DISCUSSION THREAD

**Discussion by SKT1yang** [oxc-project/oxc#5432] [Q&A] [answered]
It seems that preserve_notate_commends only retains some comments instead of all comments. What should I do if I want to preserve all comments?

The following is the code I printed. When parsing, the comment node was present, but during the codegen stage, some regular comments were deleted, even though I configured preserve_notate_commends：

```
pub fn wrap_script(source_text: &String, source_type: SourceType, language_source: &str) -> String {
    let allocator = Default::default();
    let ret = Parser::new(&allocator, &source_text, source_type).parse();
    let program = allocator.alloc(ret.program);

    let (symbols, scopes) = SemanticBuilder::new(source_text, source_type)
        .build(program)
        .semantic
        .into_symbol_table_and_scope_tree();
    let mut traverser = WrapTransformer::new(&allocator, language_source);
    traverse_mut(&mut traverser, &allocator, program, symbols, scopes);

    println!("Comments:");
    let comments = ret
        .trivias
        .comments()
        .map(|comment| comment.span.source_text(&source_text))
        .collect::<Vec<_>>();
    println!("{comments:?}");

    WrapVisitorMut::new(&allocator).build(program);

    let code = CodeGenerator::new().with_options(CodegenOptions {
        single_quote: false,
        // remove whitespace
        minify: false,
    });
    code.enable_comment(
        &source_text,
        ret.trivias.clone(),
        CommentOptions {
            preserve_annotate_comments: true,
        },
    )
    .build(program)
    .source_text
}
```

```
    #[test]
    fn comment() {
        let source_text = r#"
/* #__NO_SIDE_EFFECTS__ */ y => y
// test comment inline
        "#;
        let source_type = SourceType::default();
        let code = wrap_script(&source_text.to_string(), source_type, "language/index.ts");
        assert_eq!(code, "/* #__NO_SIDE_EFFECTS__ */ (y) => y;// test comment inline\n;");
    }
```
This is the result after running the test, not what I wanted：

<img width="664" alt="image" src="https://github.com/user-attachments/assets/73e966e1-8013-4ba7-bc88-9a50d1dde769">

**Comment by SKT1yang** [comment]
I have upgraded to the latest v0.26.0

**Comment by Boshen** [comment, marked as answer]
I'm currently working on printing comments https://github.com/oxc-project/oxc/issues/1046

No ETA so far.

**Reply by SKT1yang** [reply]
As long as this feature will supported, I can wait as long as possible;

**Comment by Boshen** [comment]
Printing all comments correctly is a really difficult task and we won't be able to support it in the near future, but we managed to print all "useful" comments".

**Reply by SKT1yang** [reply]
I can roughly understand the difficulty involved, but I have a confusion that I hope to receive an answer to. Currently, there is an oxlint tool that can retain the source code of non fixed parts when fixing, including comments. Because the lint tool requires me to understand that I should not be able to delete the correct parts written by the user. What is the difference between these two?
